sql/moz-fx-data-shared-prod/search_terms_derived/suggest_impression_sanitized_v2/query.sql

```diff
@@ -0,0 +1,81 @@
+/*
+
+This query is run in a standalone project and is scheduled separately from our
+normal Airflow infrastructure, but is codified here for discoverability and for
+making use of the testing harness here.
+
+Note that we need to fully qualify all table references since this runs from
+a separate project.
+
+*/
+WITH impressions AS (
+  SELECT
+    submission_timestamp,
+    request_id,
+    search_query AS telemetry_query,
+    advertiser,
+    block_id,
+    context_id,
+    sample_id,
+    is_clicked,
+    locale,
+    metadata.geo.country,
+    metadata.geo.subdivision1 AS region,
+    normalized_os,
+    normalized_os_version,
+    release_channel,
+    position,
+    reporting_url,
+    scenario,
+  FROM
+    `moz-fx-data-shared-prod.contextual_services_stable.quicksuggest_impression_v1`
+  WHERE
+    DATE(submission_timestamp) = @submission_timestamp
+),
+merino_logs AS (
+  SELECT
+    TIMESTAMP_TRUNC(timestamp, SECOND) AS merino_timestamp,
+    jsonPayload.fields.rid AS request_id,
+    jsonPayload.fields.query,
+    NULLIF(jsonPayload.fields.country, 'none') AS merino_country,
+    NULLIF(jsonPayload.fields.region, 'none') AS merino_region,
+    NULLIF(jsonPayload.fields.dma, 'none') AS merino_dma,
+    -- -- We are not propagating city-level data to sanitized table.
+    -- NULLIF(jsonPayload.fields.city, 'none') AS merino_city,
+    NULLIF(jsonPayload.fields.form_factor, 'none') AS merino_form_factor,
+    NULLIF(jsonPayload.fields.browser, 'none') AS merino_browser,
+    NULLIF(jsonPayload.fields.os_family, 'none') AS merino_os_family,
+  FROM
+    `suggest-searches-prod-a30f.logs.stdout`
+  WHERE
+    DATE(timestamp) = @submission_timestamp
+    AND jsonPayload.type = "web.suggest.request"
+),
+allowed_queries AS (
+  SELECT
+    query
+  FROM
+    `moz-fx-data-shared-prod.search_terms_derived.remotesettings_suggestions_v1`
+  CROSS JOIN
+    UNNEST(keywords) AS query
+),
+merino_sanitized AS (
+  SELECT
+    IF(allowed_queries.query IS NOT NULL, query, '<disallowed>') AS sanitized_query,
+    merino_logs.* EXCEPT (query)
+  FROM
+    merino_logs
+  LEFT JOIN
+    allowed_queries
+  USING
+    (query)
+)
+SELECT
+  * EXCEPT (request_id, sanitized_query, telemetry_query),
+  COALESCE(sanitized_query, telemetry_query) AS sanitized_query,
+FROM
+  impressions
+LEFT JOIN
+  merino_sanitized
+USING
+  (request_id)
```

Comment:
I think this table needs to be one row per impression, with a single merino request potentially generating several impression pings, and hence matching multiple impression rows on `request_id`. This is implicitly dropping all Merino requests that did not produce matching telemetry.

Does @rebecca-burwei have feedback on this design?

Reply:
It's ok with me to drop all Merino requests that don't match on an impression.

Yes, this is ok. In the future, we might want to reduce this to one row per `request_id` by aggregating the matching impressions into a struct keyed on `position`. But we wouldn't get much use out of this aggregation until [this other new tab telemetry work](https://docs.google.com/document/d/1ZlKOQYyhlWzlbRKW9HhdmbekKZ9W6jENXmER8IKOTKE/edit#bookmark=id.u3fya4a3qkdl) is done.

Reply:
Based on the current implementation in Firefox, each merino request would only generate at most 1 impression in Firefox. The Firefox Suggest impression ping is recorded with the following two conditions (see more details in this [doc](https://docs.google.com/document/d/1qLb4hUwR8YQj5QnjJtwxQIoDCPLQ6XuAmJPQ6_WmS4E/edit?usp=sharing)):
* there was a Firefox Suggest served at the end of the search session
* the user picked one of the search results from the search dropdown list

Firefox won't record the impressions for those "flicked-by" suggestions while the user is typing in the URL bar.

So I don't think we will have multiple impression rows corresponding to the same `request_id`.